File: sdk/node/examples/place-limit-order.ts
import { randomUUID } from "node:crypto";

import { withTradeApi } from "@limeint/trade-api";
import { OrderType, Side, TimeInForce } from "@limeint/trade-api/orders";

const secret = process.env.TRADE_API_SECRET;
const symbol = process.env.TRADE_API_SYMBOL ?? "AAPL@XNAS";

if (!secret) {
  throw new Error("Set TRADE_API_SECRET");
}

await withTradeApi({ secret }, async (api) => {
  const details = await api.auth.tokenDetails({ token: api.getToken() });
  const [accountId] = details.accountIds;
  if (details.accountIds.length !== 1 || !accountId) {
    throw new Error(
      `Expected exactly one available account; received ${details.accountIds.length}`,
    );
  }
  const order = await api.orders.placeOrder({
    accountId,
    symbol,
    quantity: { value: "1" },
    side: Side.SIDE_BUY,
    type: OrderType.ORDER_TYPE_LIMIT,
    timeInForce: TimeInForce.TIME_IN_FORCE_DAY,
    limitPrice: { value: "280.00" },
    clientOrderId: randomUUID().replaceAll("-", "").slice(0, 20),
  });

  console.log("Placed order:", order.orderId);
});

File: sdk/python/examples/place_limit_order.py
"""Place a limit order, then cancel it.

This example sends a far-from-market limit order so it will not be filled
immediately, allowing the cancel to demonstrate cleanly. Do not run against
a real account without changing the symbol/price/quantity.

Usage:
    TRADE_API_SECRET=... TRADE_API_ACCOUNT_ID=... python examples/place_limit_order.py
"""

from __future__ import annotations

import os

from google.type.decimal_pb2 import Decimal  # type: ignore[import-untyped]

from trade_api import TradeAPIClient
from trade_api.orders import (
    CancelOrderRequest,
    Order,
    OrderType,
    Side,
    TimeInForce,
)


def main() -> None:
    secret = os.environ["TRADE_API_SECRET"]
    account_id = os.environ["TRADE_API_ACCOUNT_ID"]

    with TradeAPIClient(secret=secret) as client:
        order = Order(
            account_id=account_id,
            symbol="AAPL@XNAS",
            quantity=Decimal(value="1"),
            side=Side.SIDE_BUY,
            type=OrderType.ORDER_TYPE_LIMIT,
            time_in_force=TimeInForce.TIME_IN_FORCE_DAY,
            limit_price=Decimal(value="100.00"),  # far below market
            client_order_id="example-001",
        )
        state = client.orders.PlaceOrder(order)
        print(f"Placed: {state.order_id} status={state.status}")

        cancelled = client.orders.CancelOrder(
            CancelOrderRequest(account_id=account_id, order_id=state.order_id)
        )
        print(f"Cancelled: {cancelled.order_id} status={cancelled.status}")


if __name__ == "__main__":
    main()
